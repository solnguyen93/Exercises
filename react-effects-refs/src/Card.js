import React, { useState, useEffect } from 'react';
import './Card.css';

function Card({ card }) {
    const [transform, setTransform] = useState({});

    useEffect(() => {
        const randomRotation = Math.floor(Math.random() * 81) - 40;
        const randomPositionX = Math.floor(Math.random() * 61) - 30;
        const randomPositionY = Math.floor(Math.random() * 61) - 30;

        setTransform({ rotation: randomRotation, positionX: randomPositionX, positionY: randomPositionY });
    }, [card]);

    return (
        <div className="Card">
            <div className="Card-container">
                {card ? (
                    <img
                        src={card.image}
                        alt={`${card.value} of ${card.suit}`}
                        style={{
                            transform: `translate(${transform.positionX}px, ${transform.positionY}px) rotate(${transform.rotation}deg)`,
                        }}
                    />
                ) : null}
            </div>
        </div>
    );
}

export default Card;
