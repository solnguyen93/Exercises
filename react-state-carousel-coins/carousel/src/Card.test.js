import React from 'react';
import Card from './Card.js';
import { render } from '@testing-library/react';
// import '@testing-library/jest-dom/extend-expect';

it('renders', () => {
    render(<Card />);
});

it('matches snapshot', () => {
    const { asFragment } = render(<Card />);
    expect(asFragment()).toMatchSnapshot();
});

test('matches snapshot with custom props', () => {
    const { asFragment } = render(<Card caption="Custom Caption" src="custom.jpg" currNum={1} totalNum={3} />);
    expect(asFragment()).toMatchSnapshot();
});
