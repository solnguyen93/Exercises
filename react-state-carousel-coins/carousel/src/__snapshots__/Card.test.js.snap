// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    />
    <img
      class="Card-image"
    />
    <small
      class="Card-small"
    >
      Image  of .
    </small>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot with custom props 1`] = `
<DocumentFragment>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      Custom Caption
    </h4>
    <img
      alt="Custom Caption"
      class="Card-image"
      src="custom.jpg"
    />
    <small
      class="Card-small"
    >
      Image 1 of 3.
    </small>
  </div>
</DocumentFragment>
`;
